src/scalar_impl.h

```diff
@@ -294,13 +305,130 @@ static void secp256k1_scalar_split_lambda(secp256k1_scalar *r1, secp256k1_scalar
  * Cryptography on Sensor Networks Using the MSP430X Microcontroller" (Gouvea, Oliveira, Lopez),
  * Section 4.3 (here we use a somewhat higher-precision estimate):
  * d = a1*b2 - b1*a2
- * g1 = round((2^272)*b2/d)
- * g2 = round((2^272)*b1/d)
+ * g1 = round((2^384)*b2/d)
+ * g2 = round((2^384)*(-b1)/d)
+ *
+ * (Note that 'd' is also equal to the curve order, n, here because [a1,b1] and [a2,b2]
+ * can be found as outputs of the Extended Euclidean Algorithm on inputs 'n' and 'lambda').
+ *
+ * The function below splits a in r1 and r2, such that
+ * - r1 + lambda * r2 == a (mod n)
+ * - either 0 <= r1 <= (a1 + a2 + 1) / 2 or n - (a1 + a2 + 1)/2 <= r1 < n.
+ * - either 0 <= r2 <= (-b1 + b2) / 2 or n - (-b1 + b2)/2 <= r2 < n.
+ *
+ * Proof.
+ *
+ * Let
+ *  - epsilon1 = |g1/2^384 - b2/d|
+ *  - epsilon2 = |g2/2^384 - (-b1)/d|
+ *  - c1 = round(a*g1/2^384)
+ *  - c2 = round(a*g2/2^384)
+ *
+ * Lemma 1: |c1 - a*b2/d| < 2^-1 + 2^-129
+ *
+ *    |c1 - a*b2/d|
+ *  =
+ *    |c1 - a*g1/2^384 + a*g1/2^384 - a*b2/d|
+ * <=   {triangle inequality}
+ *    |c1 - a*g1/2^384| + |a*g1/2^384 - a*b2/d|
+ *  =
+ *    |c1 - a*g1/2^384| + a*|g1/2^384 - b2/d|
+ * <=   {property of rounding in c1 & definition of epsilon1
+ *    2^-1 + a*epsilon1
+ * <    {a < 2^256 and epsilon1 < 2^-385}
+ *    2^-1 + 2^256 * 2^-385
+ *  =
+ *    2^-1 + 2^-129
```

Comment:
When using the exact value of epsilon1 here, I get 2^-10 + 2^-131.574091440504...